R/binary_classification.R

```diff
@@ -138,3 +138,210 @@ fbeta_score <- function(actual, predicted, beta = 1) {
+    cm
+}
+
+#' Sensitivity
+#'
+#' \code{sensitivity} calculates the proportion of actual positives
+#' that are correctly identified as such. It is also known as
+#' \code{true positive rate}.
+#'
+#' @inheritParams params_binary
+#' @seealso \code{\link{cmat}} \code{\link{specificity}}
+#' @export
+#' @examples
+#' actual <- c(1, 1, 1, 0, 0, 0)
+#' predicted <- c(1, 0, 1, 1, 1, 1)
+#' sensitivity(actual, predicted)
+
+sensitivity <- function(actual, predicted) {
+    cm <- cmat(actual, predicted)
+    cm$tp / (cm$tp + cm$fn)
+}
+
+#' Specificity
+#'
+#' \code{specificity} calculates the proportion of actual negatives
+#' that are correctly identified as such. It is also known as
+#' \code{true negative rate}.
+#'
+#' @inheritParams params_binary
+#' @seealso \code{\link{cmat}} \code{\link{sensitivity}}
+#' @export
+#' @examples
+#' actual <- c(1, 1, 1, 0, 0, 0)
+#' predicted <- c(1, 0, 1, 1, 1, 1)
+#' specificity(actual, predicted)
+
+specificity <- function(actual, predicted) {
+    cm <- cmat(actual, predicted)
+    cm$tn / (cm$tn + cm$fp)
+}
+
+#' False Negative Rate
+#'
+#' \code{fnr} calculates the proportion of actual positives
+#' that are not identified as such.
+#' It is defined as \code{1 - sensitivity}
+#'
+#' @inheritParams params_binary
+#' @export
+#' @seealso \code{\link{cmat}} \code{\link{sensitivity}}
+#' @examples
+#' actual <- c(1, 1, 1, 0, 0, 0)
+#' predicted <- c(1, 0, 1, 1, 1, 1)
+#' fnr(actual, predicted)
+
+fnr <- function(actual, predicted) {
+    1 - sensitivity(actual, predicted)
+}
+
+#' False Positive Rate
+#'
+#' \code{fnr} calculates the proportion of actual negatives
+#' that are not identified as such.
+#' It is defined as \code{1 - specificity}
+#'
+#' @inheritParams params_binary
+#' @seealso \code{\link{cmat}} \code{\link{specificity}}
+#' @export
+#' @examples
+#' actual <- c(1, 1, 1, 0, 0, 0)
+#' predicted <- c(1, 0, 1, 1, 1, 1)
+#' fpr(actual, predicted)
+
+fpr <- function(actual, predicted) {
+    1 - specificity(actual, predicted)
+}
+
+#' Positive Predictive Value
+#'
+#' \code{ppv} calculates the proportion positive values that
+#' are true positive results.
+#'
+#' @inheritParams params_binary
+#' @seealso \code{\link{cmat}} \code{\link{npv}}
+#' @export
+#' @examples
+#' actual <- c(1, 1, 1, 0, 0, 0)
+#' predicted <- c(1, 0, 1, 1, 1, 1)
+#' ppv(actual, predicted)
+
+ppv <- function(actual, predicted) {
+    cm <- cmat(actual, predicted)
+    cm$tp / (cm$tp + cm$fp)
+}
+
+#' Negative Predictive Value
+#'
+#' \code{ppv} calculates the proportion negative values that
+#' are true negative results.
+#'
+#' @inheritParams params_binary
+#' @seealso \code{\link{cmat}} \code{\link{npv}}
+#' @export
+#' @examples
+#' actual <- c(1, 1, 1, 0, 0, 0)
+#' predicted <- c(1, 0, 1, 1, 1, 1)
+#' npv(actual, predicted)
+npv <- function(actual, predicted) {
+    cm <- cmat(actual, predicted)
+    cm$tn / (cm$tn + cm$fn)
+}
+
+#' False Discovery Rate
+#'
+#' \code{fdr} is the complement of the Positive Predictive
```

Comment:
I'd prefer to be a bit more explicit with the documentation for these functions. For example, I'd write something like

```
\code{fdr} computes proportion of observations predicted to be in the positive class (i.e. the element in \code{predicted} equals 1) that actually belong to the negative class (i.e. the element in \code{actual} equals 0).
```

Does that make sense? I just think that the terms `positive results` or `false positives` in the proposed documentation may be a bit confusing to people that have never seen these functions before.

Reply:
Also, can you move the language that fdr is `1 - positive_predicted_value` to the `Details` section of the documentation?

Reply:
Done with some documentation. Will review it again in a few hours to make sure the docs are clear and complete.